### PR TITLE
LogBuffering backend

### PR DESCRIPTION
**Pull request by denisshevchenko**
description
-----------

- [x] `LogBuffer` keeps the latest LogObject of a logger name; can output via `Switchboard.readLogBuffer`; added to `example-complex`

checklist
---------

- [x] compiles (`cabal new-clean; cabal new-build`)
- [x] tests run successfully (`cabal new-test`)
- [ ] documentation added and created (`cd docs; nix-shell --run make`)
- [x] link to an issue
- [x] link to an epic
- [x] add estimate points
- [x] add milestone (the same as the linked issue)

### CODE REVIEW

**Comment by CodiePP** [comment]
the `example-complex` will output the aggregation of the random numbers every 25 seconds. :-)

File: iohk-monitoring/src/Cardano/BM/Output/LogBuffering.lhs [hunk 1]
```diff
@@ -0,0 +1,83 @@
+\subsection{Cardano.BM.Output.LogBuffer}
+\label{module:Cardano.BM.Output.LogBuffer}
```
Comment (365andreas) [inline iohk-monitoring/src/Cardano/BM/Output/LogBuffering.lhs]: This file shouldn't be deleted? It is the same with `iohk-monitoring/src/Cardano/BM/Output/LogBuffer.lhs`.

File: iohk-monitoring/src/Cardano/BM/Output/Switchboard.lhs [hunk 2]
```diff
@@ -264,67 +274,78 @@ instance (Show a, ToJSON a) => IsBackend Switchboard a where
...
-setupBackends (bk : bes) c sb acc = do
-    be' <- setupBackend' bk c sb
-    setupBackends bes c sb ((bk,be') : acc)
-setupBackend' :: (Show a, ToJSON a) => BackendKind -> Configuration -> Switchboard a -> IO (Backend a)
+setupBackends bes c sb = catMaybes <$>
+                         forM bes (\bk -> do { setupBackend' bk c sb >>= \case Nothing -> return Nothing
+                                                                               Just be -> return $ Just (bk, be) })
+
+setupBackend' :: (Show a, ToJSON a) => BackendKind -> Configuration -> Switchboard a -> IO (Maybe (Backend a))
 setupBackend' SwitchboardBK _ _ = error "cannot instantiate a further Switchboard"
```
Comment (365andreas) [inline iohk-monitoring/src/Cardano/BM/Output/Switchboard.lhs]: Over 100 chars width.

Comment (CodiePP) [reply iohk-monitoring/src/Cardano/BM/Output/Switchboard.lhs]: :-) works on large screen ...

Comment (denisshevchenko) [reply iohk-monitoring/src/Cardano/BM/Output/Switchboard.lhs]: Should we buy a wide monitor? :-D

File: iohk-monitoring/src/Cardano/BM/Output/LogBuffering.lhs (hunk 1 above)
Comment (CodiePP) [reply iohk-monitoring/src/Cardano/BM/Output/LogBuffering.lhs]: yes.

**Pull request merged**